lib/events.js

```diff
@@ -497,6 +497,17 @@ function unwrapListeners(arr) {
 
 function once(emitter, name) {
   return new Promise((resolve, reject) => {
+    if (typeof emitter.addEventListener === 'function') {
+      // EventTarget does not have `error` event semantics like Node
+      // EventEmitters, we do not listen to `error` events here.
+      emitter.addEventListener(
+        name,
+        (...args) => { resolve(args); },
```

Comment:
EventTarget is guaranteed to only have one argument, so you could avoid passing an array here.

Reply:
@domenic what do you think is the right call API wise in terms of compatibility with EventEmitter and the coming (?) events proposal?

Reply:
I wouldn't pay much attention to early-stage TC39 proposals. But I think just compatibility with how events work in event listeners in the DOM would suggest using a single argument instead of an array containing that argument.

Reply:
So this now resolves with an array, and just got released in Node v12.11.0. Are we stuck with this?

Reply:
> So this now resolves with an array, and just got released in Node v12.11.0. Are we now stuck with this promise signature?

I think changing that would be a breaking change, yes.

That being said, I think there is some value in keeping this consistent between EventTarget and EventEmitter, so I’m not really sure that we *should* change this even if we could.

Reply:
What about consistency with web EventTarget? If that is a goal, arrays should not get involved.

Reply:
@domenic @addaleax I thought about return one argument, but for consistency with `EventEmitter` decide to keep the array.

I will open soon a new PR with some improvements.
We can go with `EventTarget` API and I will update the code, docs and the tests.

WDYT?

Reply:
> What about consistency with web EventTarget? If that is a goal, arrays should not get involved.

@domenic Yes, there’s a tradeoff, but at least to me a polymorphic API seems worse here than one that performs some unnecessary boxing.

That being said: I personally don’t like the decision to use arrays in the first place, but I think we’re stuck with that now.

Reply:
FWIW I'm looking at this and I'm pretty sure Domenic was right and our implementation is wrong. Even for consistency. 

Our tests incorrectly use an EventTargetMock that accepts multiple arguments (which isn't possible) and dispatches strings (and not objects).

Reply:
Fixed in #33659